### ISSUE TITLE
[no-unused-vars] false positive with `@link` and `import type`

### ISSUE THREAD

**Issue by arobinson** [typescript-eslint/typescript-eslint#4088]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
'@typescript-eslint/no-unused-vars': ['error', { vars: 'all', args: 'all', ignoreRestSiblings: false, argsIgnorePattern: '^_' }],
```

```TS
import type { RecordingState } from '../nw-recording/nw-recording';
...
@Injectable()
export abstract class NwAutomationRecordingService {
  abstract readonly recordingInOtherWindow$: Observable<boolean>;

  /**
   * Returns if recording is active in another window. Recording in the current window is disabled, but the recording state will still
   * be {@link RecordingState.NotRecording}
   */
  abstract isRecordingInOtherWindow(): boolean;
...
}
```

```JSON
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "preserveSymlinks": true,
    "importHelpers": true,
    "target": "es2017",
    "esModuleInterop": false,
    "types": ["jasmine", "node"],
    "typeRoots": ["node_modules/@types"],
    "lib": ["ES2020", "dom"],
    "removeComments": true,
    "pretty": true,
    "downlevelIteration": true,
    "noImplicitReturns": true,
    "strict": false,
    "paths": {
      "nw-shared": ["./nw-shared-src/public-api.ts"]
    }
  },
  "angularCompilerOptions": {
    "preserveWhitespaces": false,
    "strictInjectionParameters": true,
    "strictInputAccessModifiers": true,
    "strictTemplates": true,
    "fullTemplateTypeCheck": true
  }
}
```

**Expected Result**

Since the imported type is used (and unfortunately needed) to resolve the `@link` in the TS comment for VS code, it should not be reported as unused

**Actual Result**

Rule flagged the `import type` line as an error due to it thinking it wasn't used

**Additional Info**

No

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.2` |
| `@typescript-eslint/parser`        | `4.28.2` |
| `TypeScript`                       | `4.3.4` |
| `ESLint`                           | `7.30.0` |
| `node`                             | `14.17.1` |

**Comment by bradzacher** [comment]
We do not parse or handle JSDoc comments - we only handle references in code, not comments.

If you want to maintain a link, instead I believe you can use import types in your JSDoc
